p2p/host/basic/basic_host.go

```diff
@@ -23,6 +23,8 @@ var log = logging.Logger("basichost")
 
 var NegotiateTimeout = time.Second * 60
 
+type AddrsFactory func([]ma.Multiaddr) []ma.Multiaddr
+
 // Option is a type used to pass in options to the host.
```

Comment:
I have a bit of work-in-progress code to refactor construction of BasicHost and Swarm, the Option type will be gone afterwards. Separate PR though :)